applications/GeoMechanicsApplication/custom_elements/U_Pw_small_strain_FIC_element.cpp

```diff
@@ -463,24 +451,21 @@ void UPwSmallStrainFICElement<TDim, TNumNodes>::CalculateAll(MatrixType& rLeftHa
     const auto integration_coefficients =
         this->CalculateIntegrationCoefficients(IntegrationPoints, Variables.detJContainer);
     const auto deformation_gradients = this->CalculateDeformationGradients();
-    const auto determinants_of_deformation_gradients =
-        GeoMechanicsMathUtilities::CalculateDeterminants(deformation_gradients);
-    const auto strain_vectors = this->CalculateStrains(
+    auto       strain_vectors        = this->CalculateStrains(
         deformation_gradients, b_matrices, Variables.DisplacementVector, Variables.UseHenckyStrain);
+    std::vector<Matrix> constitutive_matrices;
+    this->CalculateAnyOfMaterialResponse(deformation_gradients, ConstitutiveParameters,
+                                         Variables.NContainer, Variables.DN_DXContainer,
+                                         strain_vectors, mStressVector, constitutive_matrices);
 
     // Loop over integration points
     for (unsigned int GPoint = 0; GPoint < NumGPoints; ++GPoint) {
         // Compute Np, GradNpT, B and StrainVector
         this->CalculateKinematics(Variables, GPoint);
-        Variables.B = b_matrices[GPoint];
-
-        // Compute infinitessimal strain
-        Variables.F            = deformation_gradients[GPoint];
-        Variables.detF         = determinants_of_deformation_gradients[GPoint];
-        Variables.StrainVector = strain_vectors[GPoint];
-
-        // set gauss points variables to constitutivelaw parameters
-        this->SetConstitutiveParameters(Variables, ConstitutiveParameters);
+        Variables.B                  = b_matrices[GPoint];
+        Variables.F                  = deformation_gradients[GPoint];
```

Comment:
Here I feel that it is not so obvious that we won't need `Variables.detF` anymore, since `Variables` is still being passed around a lot. Just to double-check: are you sure we can remove it here safely? (For `InitializeNonLinearIteration` and `FinalizeNonLinearIteration` you have convinced me that it is safe to do so.)

Reply:
The only place where detF for the UPwSmallStrain route is still used (aside from assigning it), is in the SetConstitutiveParameters function. The SetConstitutiveParameters function is only used in 2 places (`InitializeSolutionStep` and `FinalizeSolutionStep` of the `UPwSmallStrainElement`), which are both not related to this part and in both these cases the detF is kept.

Reply:
And this is the same for the diff order route.